fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidateOnlyTest.java

```diff
@@ -133,18 +133,20 @@ void testValidateMissingSplit(int splitNumber, FormatVersion formatVersion, bool
                         ValidationTestUtils.assertInvalidResults(
                                 invalidResults,
                                 1,
-                                result -> result.equals(RecordRepairResult.invalid(primaryKey, RecordRepairResult.CODE_SPLIT_ERROR, "any")));
+                                result -> !result.isValid() &&
```

Comment:
My comment here: https://github.com/FoundationDB/fdb-record-layer/pull/3370#discussion_r2136336731
Was about the amount of repetition. I _think_ you should be able to have a helper method that takes a few arguments and covers all of these assertions. This would make it much clearer, what the different expectations are. 
A single `ValidationTestUtils.assertResults` that takes both the `repairStats` and `repairResults`, could (I think) cover `assertRepairStats`, `assertCompleteResults`, `assertInvalidResults`. That method may want to use the existing methods in its implementation.

I mostly brought this up because you noted a feeling that these tests are rather long and complicated, and this could help reduce the cognitive complexity of the assertions.